cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishStartService.java

```diff
@@ -0,0 +1,155 @@
+	private final PublishConfiguration publishConfiguration;
+	private final PublishExecutor publishExecutor;
+	private final PublishJobFactory jobFactory;
+
+	private static final Logger logger = LoggerFactory.getLogger(PublishStartService.class);
+
+	@Inject
+	public PublishStartService(
+			PublishWebhookCommandHandler publishWebhookCommandHandler,
+			CmsItemLookupReporting lookupReporting,
+			PublishConfiguration publishConfiguration,
+			PublishExecutor publishExecutor,
+			PublishJobFactory jobFactory) {
+
+		this.publishWebhookCommandHandler = publishWebhookCommandHandler;
+		this.lookupReporting = lookupReporting;
+		this.publishConfiguration = publishConfiguration;
+		this.publishExecutor = publishExecutor;
+		this.jobFactory = jobFactory;
+	}
+
+
+	public LinkedHashMap<String, String> doPublishStartItem(CmsItemId itemId, PublishStartOptions options) {
+
+		PublishConfig config = getPublishConfiguration(itemId, options.getPublication());
+		PublishJob publishJob = getPublishJob(itemId, options, config);
+
+		Set<String> uuids = publishExecutor.startPublishJobs(new LinkedHashSet<PublishJob>(Arrays.asList(publishJob)));
+
+		// Verify that one job was started, log the UUID.
+		uuids.forEach(uuid -> logger.info("Publish job started: {}", uuid));
+		if (uuids.size() != 1) {
+			throw new CommandRuntimeException("StartPublishJobError", String.format("None or several publish jobs started: %d", uuids.size()));
+		}
+
+		PublishJobOptions publishJobOptions = publishJob.getOptions();
+		PublishJobDelivery delivery = publishJobOptions.getDelivery();
+		PublishJobStorage storage = publishJobOptions.getStorage();
+		Optional<PublishJobProgress> progress = Optional.of(publishJobOptions.getProgress());
+
+		if (storage == null) {
+			throw new IllegalArgumentException("Need a valid PublishJobStorage object with params archive and manifest.");
+		}
+
+		logger.debug("WebhookCommandHandler endpoint: {}", delivery.getParams().get("url"));
+		LinkedHashMap<String, String> result = publishWebhookCommandHandler.getPostPayload(delivery, storage, progress);
+
+		return result;
+	}
+
+	private PublishConfig getPublishConfiguration(CmsItemId itemId, String name) {
+
+		Map<String, PublishConfig> configs = publishConfiguration.getConfiguration(itemId);
+		return configs.get(name);
+	}
+
+	private PublishJob getPublishJob(CmsItemId itemId, PublishStartOptions options, PublishConfig config) {
+
+		CmsItem item = this.lookupReporting.getItem(itemId);
+		CmsItemPublish itemPublish = new CmsItemPublish(item);
+		TranslationLocalesMapping localesRfc = (TranslationLocalesMapping) this.publishConfiguration.getTranslationLocalesMapping(itemPublish);
+
+		PublishProfilingRecipe profilingRecipe = null;
+		// Use profilingname if set (get recipe from itemPublish)
+		if (itemPublish.hasProfiles() && options.getProfilingname() != null && options.getProfilingname().length() > 0) {
+			ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Don't remove the injected ObjectMapper...

Reply:
Most preferable is injecting the higher level ObjectReader.

Reply:
Done.